artemis-core-client/src/main/java/org/apache/activemq/artemis/spi/core/remoting/ssl/SSLContextConfig.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2021 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.spi.core.remoting.ssl;
+
+import java.util.Objects;
+
+import org.apache.activemq.artemis.core.remoting.impl.netty.TransportConstants;
+
+/**
+ * This class holds configuration parameters for SSL context initialization.
+ * To be used with {@link SSLContextFactory} and {@link OpenSSLContextFactory}.
+ */
+public final class SSLContextConfig {
+
+   public static final class Builder {
```

Comment:
This `Builder` class seems a bit redundant to me. It has all the same properties as the parent `SSLContextConfig`. Why not just make `SSLContextConfig` fluent by having normal setter methods which return `this` (as is done for other configuration objects in the code-base, e.g. `org.apache.activemq.artemis.core.config.DivertConfiguration`)?

Reply:
This way SSLContext is immutable and can be directly used as cache key in the caching factories.

Reply:
Makes sense, @sebthom. Thanks for the explanation.